// react-native-document-scanner-plugin/lib/module/index.js
import { NativeModules, Platform } from 'react-native';
const LINKING_ERROR = `The package 'react-native-document-scanner-plugin' doesn't seem to be linked. Make sure: \n\n` + Platform.select({
  ios: "- You have run 'pod install'\n",
  default: ''
}) + '- You rebuilt the app after installing the package\n' + '- You are not using Expo managed workflow\n';
const DocumentScanner = NativeModules.DocumentScanner ? NativeModules.DocumentScanner : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }
});
export default {
  /**
   * Opens the camera, and starts the document scan
   */
  scanDocument() {
    let options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
    return DocumentScanner.scanDocument(options);
  }
};
export let ResponseType;
(function (ResponseType) {
  ResponseType["Base64"] = "base64";
  ResponseType["ImageFilePath"] = "imageFilePath";
})(ResponseType || (ResponseType = {}));
export let ScanDocumentResponseStatus;
(function (ScanDocumentResponseStatus) {
  ScanDocumentResponseStatus["Success"] = "success";
  ScanDocumentResponseStatus["Cancel"] = "cancel";
})(ScanDocumentResponseStatus || (ScanDocumentResponseStatus = {}));
